app/jobs/invoices/abacus/create_yearly_invoices_job.rb

```diff
@@ -15,22 +15,52 @@ class Invoices::Abacus::CreateYearlyInvoicesJob < BaseJob
 
   def initialize(invoice_year:, invoice_date:, send_date:, role_finish_date:)
     super()
+    raise ArgumentError, "invoice_year must be a positive integer" unless invoice_year.positive?
     @invoice_year = invoice_year
     @invoice_date = invoice_date
     @send_date = send_date
     @role_finish_date = role_finish_date
   end
 
+  def enqueue!(options = {})
+    assert_no_other_job_running!
+    super
+  end
+
   def perform
-    # abort_if_other_job_is_running
+    assert_no_other_job_running!
     # log start according to ticket
-    # extend_roles_for_invoicing
+    extend_roles_for_invoicing
     # process_invoices
     # log finish according to ticket
   end
 
+  def active_members
+    Person
+      .where.not(abacus_subject_key: nil)
+      .where.not(data_quality: :error)
+      .joins(:roles)
+      .merge(Role.active(Date.new(@invoice_year)))
+      .where(roles: {type: Group::SektionsMitglieder::Mitglied.sti_name})
+      .left_outer_joins(:external_invoices)
+      .where("external_invoices.id IS NULL OR external_invoices.type != ? OR external_invoices.year != ?", ExternalInvoice::SacMembership.sti_name, @invoice_year)
+  end
+
   private
 
+  def assert_no_other_job_running!
+    raise "There is already a job running" if other_job_running?
+  end
+
+  def other_job_running?
+    Delayed::Job.where("handler LIKE ?", "%Invoices::Abacus::CreateYearlyInvoicesJob%")
+      .where(failed_at: nil).exists?
+  end
```

Comment:
Das könnte, wenn via `perform` aufgerufen, true zurück geben, wenn der eigene Job gefunden wird. Kann wohl via `delayed_jobs.where.not(id: @delayed_job.id)`  (siehe BaseJob) gelöst werden.

Reply:
Ja, nur beim enque passt.

Reply:
Fixed.